pkg/kube/controllers/extendedjob/job_reconciler.go

```diff
@@ -178,27 +185,50 @@ func (r *ReconcileJob) persistOutput(ctx context.Context, instance *batchv1.Job,
 
 		// Create secret and persist the output
 		secretName := conf.NamePrefix + c.Name
-		secret := &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      secretName,
-				Namespace: instance.GetNamespace(),
-				Labels:    conf.SecretLabels,
-			},
-			StringData: data,
-		}
 
-		err = r.client.Get(ctx, types.NamespacedName{Name: secretName, Namespace: instance.GetNamespace()}, &corev1.Secret{})
-		if apierrors.IsNotFound(err) {
-			err = r.client.Create(ctx, secret)
+		if conf.ToBeVersioned {
```

Comment:
We don't want the `ExtendedJob` to do versioning.
It's fine if the variable interpolation `ExtendedJob` saves output to a normal secret, and the **manifest reconciler** picks that up, and it then versions it.

We don't want any of the `Extended*` resources to be dependent on BOSH concepts.

Reply:
So what **manifest reconciler** does is that watch new desired manifest from var-interpolation job and store as versioned manifest. So it just archives each manifest and **does not** apply versioned manifest to data-gathering job input. Is it right?